src/Node/HTTP/HTTPS.purs

```diff
@@ -0,0 +1,242 @@
+cert :: Option SSLOptions Cert
+cert = cmap extract $ opt "cert"
+  where
+    extract (CertString s) = toForeign s
+    extract (CertBuffer b) = toForeign b
+    extract (CertStringArray sa) = toForeign sa
+    extract (CertBufferArray ba) = toForeign ba
+
+-- | The CA option can be a String, a Buffer, an array of strings, or an array
+-- | of buffers.
+data CA
+  = CAString String
+  | CABuffer Buffer
+  | CAStringArray (Array String)
+  | CABufferArray (Array Buffer)
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+ca :: Option SSLOptions CA
+ca = cmap extract $ opt "ca"
+  where
+    extract (CAString s) = toForeign s
+    extract (CABuffer b) = toForeign b
+    extract (CAStringArray sa) = toForeign sa
+    extract (CABufferArray ba) = toForeign ba
+
+-- | The CRL option can be a String, a Buffer, an array of strings, or an array
+-- | of buffers.
+data CRL
+  = CRLString String
+  | CRLBuffer Buffer
+  | CRLStringArray (Array String)
+  | CRLBufferArray (Array Buffer)
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+crl :: Option SSLOptions CRL
+crl = cmap extract $ opt "crl"
+  where
+    extract (CRLString s) = toForeign s
+    extract (CRLBuffer b) = toForeign b
+    extract (CRLStringArray sa) = toForeign sa
+    extract (CRLBufferArray ba) = toForeign ba
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+ciphers :: Option SSLOptions String
+ciphers = opt "ciphers"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+honorCipherOrder :: Option SSLOptions Boolean
+honorCipherOrder = opt "honorCipherOrder"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+ecdhCurve :: Option SSLOptions String
+ecdhCurve = opt "ecdhCurve"
+
+-- | The DHParam option can take either a String or a Buffer
+data DHParam = DHParamString String | DHParamBuffer Buffer
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+dhparam :: Option SSLOptions DHParam
+dhparam = cmap extract $ opt "dhparam"
+  where
+    extract (DHParamString s) = toForeign s
+    extract (DHParamBuffer b) = toForeign b
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+secureProtocol :: Option SSLOptions String
+secureProtocol = opt "secureProtocol"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+secureOptions :: Option SSLOptions Int
+secureOptions = opt "secureOptions"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)
+sessionIdContext :: Option SSLOptions String
+sessionIdContext = opt "sessionIdContext"
+
+-- | Create an HTTPS server, given the SSL options and a function to be executed
```

Comment:
No need for a comment here I think, since this isn't being exported, and people can look at `createServer` a few lines down.